_posts/2017-12-21-code-reviews.md

```diff
@@ -0,0 +1,71 @@
+---
+layout: post
+title: Why do a code review?
+author: jake_hall
+external_url: http://blog.florxlabs.com/code-reviews
+---
+
+Code reviews are an integral part of the teams that I work on, and recently I was asked what value they added versus the time they took. Here are a few thoughts about why they are absolutely vital to any software development project.
+
+# What do I mean by a code review? #
+
+To us, a code review is the final check by the team that the change being delivered has the following attributes, before being merged into the main branch: 
+* there are no obvious issues with logic
+* follows agreed best practice, and coding standards
+* code is understandable, maintainable and clean
+* all automated tests are written, and have the necessary coverage
+* the acceptance criteria on the ticket has been met
+
+# How do we do code reviews? #
+
+In order to gain the most from our code reviews, we try and stick the following: 
+* Two other people to review each pull request (PR)
+* Picking reviewers that will have the most impact
+* Adding comments to the PR on anything you think needs explaining
+* Not being defensive when receiving comments on own code (See [you are not the code you write](https://hackernoon.com/you-are-not-the-code-you-write-e54f02876ca3))
+* Always aim to resolve all review comments, if they improve the project as a whole, no matter how trivial/minor.
+* Be concerned if we get no comments, PRs are rarely perfect first time!
+
+# Why are they so important? #
+
+## Prevent Issues ##
+
+Code reviews are an additional gate, and more pairs of eyes on a change. This can increase the likelihood of picking up issues before they out of the development environment where fixing is much quicker and cheaper. At this point the developer is unlikely to have context switched too, so can make the change relatively easily.
```

Comment:
Should this be "...before they *are* out of the development environment..."?

Reply:
well spotted - added in, thanks 😁